code/controllers/subsystems/initialization/codex_dump.dm

```diff
@@ -0,0 +1,204 @@
+	sidebar += "<hr/>"
+	sidebar += "<form action='/url' method='GET'>"
+	sidebar += "<input type='text' placeholder='Search the codex.'>"
+	sidebar += "<button>Go!</button>"
+	sidebar += "</form>"
+*/
+
+	sidebar += "<hr/>"
+	sidebar += "<ul>"
+	var/list/codex_categories = decls_repository.get_decls_of_subtype(/decl/codex_category)
+	for(var/cat_type in codex_categories)
+		var/decl/codex_category/cat = codex_categories[cat_type]
+		var/datum/codex_entry/cat_entry = SScodex.get_entry_by_string("[cat.name] (category)")
+		if(cat_entry)
+			sidebar += "<li><a href='[convert_filename(cat_entry.get_dump_link_name())]'>[cat.name]</a></li>"
+	sidebar += "</ul>"
+
+	sidebar += "<hr/>"
+	sidebar += "<ul>"
+	var/githuburl = get_config_value(/decl/config/text/githuburl)
+	if(githuburl)
+		sidebar += "<li><a href='[githuburl]'>Github repository</a></li>"
+	var/discordurl = get_config_value(/decl/config/text/discordurl)
+	if(discordurl)
+		sidebar += "<li><a href='[discordurl]'>Discord community</a></li>"
+	sidebar += "</ul>"
+
+	sidebar = jointext(sidebar, "\n")
+
+/decl/codex_page_converter/proc/get_sidebar()
+	if(!sidebar)
+		generate_sidebar()
+	return sidebar
+
+// Strips out some extraneous <br> in the codex strings.
+/decl/codex_page_converter/proc/convert_body(var/title, var/list/body)
+	. = "<body>\n<div class = 'dumpCodexPage'>\n<div class = 'dumpCodexSidebar'>\n[get_sidebar()]\n</div>\n<div class = 'dumpCodexText'>\n<h1>[title]</h1>\n[jointext(body, "\n")]\n"
+	. = replacetext(., "<p><br>",  "\n<p>")
+	. = replacetext(., "<br><p>",  "\n<p>")
+	. = replacetext(., "</p><br>", "</p>\n")
+	. = replacetext(., "<br></p>", "</p>\n")
+	. = insert_header(title, .)
+	. = insert_footer(title, .)
+	. = "<!DOCTYPE html>\n<html>\n[.]\n</div>\n</div>\n</body>\n</html>"
+
+/decl/codex_page_converter/proc/strikethrough(var/thing)
+	return "<font color = '#ff0000'><s>[thing]</s></font>"
+
+/decl/codex_page_converter/proc/convert_link(var/address, var/thing)
+	return "<a href='[url_encode(address)]'>[thing]</a>"
+
+/datum/controller/subsystem/codex/proc/dump_to_filesystem(var/codex_page_converter = /decl/codex_page_converter)
+
+	var/decl/codex_page_converter/convert = GET_DECL(codex_page_converter)
+
+	// Build a reference list of filenames to datums so we can
+	// crosslink them when generating the text body. Also collect
+	// our body text in the process.
+	var/list/address_to_body =  list()
+	var/list/address_to_entry = list()
+	var/list/entry_to_address = list()
+	//var/list/seen_name_count =  list()
+	for(var/datum/codex_entry/codex_entry as anything in all_entries)
+		var/address = convert.convert_filename(codex_entry.get_dump_link_name())
+		address_to_entry[address] = codex_entry
+		entry_to_address[codex_entry] = address
+		address_to_body[address] = convert.convert_body(codex_entry.name, codex_entry.get_codex_body(include_header = FALSE, include_footer = FALSE))
+
+	// Copied from del-the-world UT exceptions list.
```

Comment:
not quite. you do all subtypes of `/atom/movable`, while dtw only instantiates `typesof(/obj/item, /obj/effect, /obj/structure, /obj/machinery, /obj/vehicle, /mob) - ignore`. for example, `/obj/random` is excluded implicitly from dtw (maybe uh, bit of an oversight there actually?). excluding it here would fix the runtime, i've got no clue why missiles are failing GC though